### PR TITLE
Make filter and post-filter behavior clearer in query builder

### LINKED ISSUE #1995: Feature request: Improve Query Builder UX
### Similar requests

- [X] I have searched and found no existing similar requests

### How are you using Legend?

_No response_

### What problems are you trying to solve?

General improvements for user experience of Query Builder 

### Describe the solution you would like to see implemented

- [x] Better behavior of negative numbers in filter
   - [x] Allow entry of negative numbers easily in filter
   - [x] Allow entry of negative numbers easily in post-filter
   - [x] Save entry of negative numbers after 'Edit Query in PURE' in filter
   - [x] Save entry of negative numbers after 'Edit Query in PURE' in post-filter
- [x] Consolidate actions in filter / + logs
- [x] Consolidate actions in post-filter / + logs
- [x] Limit DND Options in projection columns to be only when user clicks on the dragicon
- [x] +logs for (View project and View SDLC Project) 

- [x] Better save query action
   - [x] Have a Save button and a Save As button in query panel actions 
   - [x] Have an option to rename query 
   - [x] Upon Save As, have naming query input as 'Copy of [QUERY_NAME]' 
   - [x] Upon Save As, if 'Copy of [QUERY_NAME]' is the name of an existing query, caution the name input in yellow
   - [x] Disable Save button during the process of creating query from scratch 
   - [x] Relocate query name 
   - [x] Have it only change URL on save as and not reload whole thing 
   - [x] grey out the register service/ productionize buttons if query is not saved but add text of 'save your query first, before registering it as a dev service' / 'save your query first before productinizing query' respectively as hover title
   - [x] ~~Add new query button next to 'save as' button~~

- [x] Upon clicking save or save as query, do not reset the UI of the query results 

- [ ] Better help options for Query internally
   - [x] add extension mechanism in help options menu to have a list with these elements  
   - [ ] move see documentation as well as help options layer outside to top query header 
   - [ ] open support ticket link 
   - [ ] review documentation 
   - [ ] link to FAQs 
   - [ ] Contact Support (delegate help to data space owner) 

- [x] Better humanized properties 

- [ ] DND Filter 
   - [ ] user can drag attribute onto blank option from filter section and not just fetch/property explorer 
   - [ ] remove create group/create logical group/cleanup tree button actions if filter properties become drag and droppable to simplify need for blank conditions and creating/cleaning conditions 
   
- [x] Allow holding multiple rows/columns in query results to filter for multiple result conditions

General Thoughts: 
- Limit modals when we can
- Remove excess of empty space in modals
Related: #2009 
### Describe alternatives you have considered

_No response_

### Documentation, Design, Adoption, Migration Strategy

_No response_

### Contribution

- [X] I would like to work on this feature
-

### PR DESCRIPTION
**Pull request by xannem**
## Summary

fixes #1995
Kapture 2023-05-14 at 14 34 39

Kapture 2023-05-14 at 14 59 40

- [x] allows drag and drop capabilities in post-filter panel by dragging the conditions themselves  
- [x] allows drag and drop capabilities in filter panel by dragging the conditions themselves  

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings) (ux changes)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

**Review by gayathrir11 — DISMISSED** [review]

File: packages/legend-art/src/panel/BlankPanelPlaceholder.tsx [hunk 1]
```diff
@@ -68,6 +68,7 @@ export const BlankPanelPlaceholder: React.FC<{
    * corresponding to the drag-over entry, signaling dropability.
    */
   isDropZoneActive?: boolean | undefined;
+  draggableActive?: boolean | undefined;
```
Comment (akphi) [inline packages/legend-art/src/panel/BlankPanelPlaceholder.tsx:71]: --> `showDropableSuggestion`

File: packages/legend-art/src/panel/PanelDropZone.tsx [hunk 2]
```diff
@@ -30,10 +30,16 @@ export const PanelDropZone: React.FC<{
   children: React.ReactNode;
   isDragOver: boolean;
   dropTargetConnector: ConnectDropTarget;
+  draggableActive?: boolean | undefined;
```
Comment (akphi) [inline packages/legend-art/src/panel/PanelDropZone.tsx:33]:
```suggestion
  showDropableSuggestion?: boolean | undefined;
```

File: packages/legend-art/src/panel/PanelDropZone.tsx [hunk 3]
```diff
@@ -30,10 +30,16 @@ export const PanelDropZone: React.FC<{
...
 }> = (props) => {
-  const { children, isDragOver, dropTargetConnector } = props;
+  const { children, isDragOver, draggableActive, dropTargetConnector } = props;
   return (
-    <div className="dnd__dropzone" ref={dropTargetConnector}>
+    <div
+      className={clsx('dnd__dropzone', {
+        'draggable--active': draggableActive,
```
Comment (akphi) [inline packages/legend-art/src/panel/PanelDropZone.tsx:39]:
```suggestion
        'dnd__dropzone--dropable': draggableActive,
```

File: packages/legend-query-builder/src/components/fetch-structure/QueryBuilderPostFilterPanel.tsx [hunk 4]
```diff
@@ -359,80 +386,177 @@ const QueryBuilderPostFilterConditionEditor = observer(
...
+          <div
+            className={clsx(
+              'dnd__entry-potential__dropzone__indicator dnd__entry-potential__dropzone__indicator--full',
+              {
+                'dnd__entry-potential__dropzone__indicator--dragover':
+                  isDragOver,
+              },
+            )}
+          >
+            {' '}
```
Comment (akphi) [inline packages/legend-query-builder/src/components/fetch-structure/QueryBuilderPostFilterPanel.tsx:416]: let's remove these

File: packages/legend-query-builder/src/components/fetch-structure/QueryBuilderPostFilterPanel.tsx [hunk 5]
```diff
@@ -359,80 +386,177 @@ const QueryBuilderPostFilterConditionEditor = observer(
       cleanUpReloadValues,
     };
 
+    // svp0
```
Comment (akphi) [inline packages/legend-query-builder/src/components/fetch-structure/QueryBuilderPostFilterPanel.tsx:389]: can we do a quick scan again to clean these up? Thanks!

File: packages/legend-art/src/panel/BlankPanelPlaceholder.tsx (hunk 1 above)
Comment (akphi) [reply packages/legend-art/src/panel/BlankPanelPlaceholder.tsx:71]: can we document this flag to differentiate it from the `isDropZoneActive` flag?

File: packages/legend-query-builder/src/components/filter/QueryBuilderFilterPanel.tsx [hunk 6]
```diff
@@ -85,12 +85,26 @@ import {
   QUERY_BUILDER_VARIABLE_DND_TYPE,
 } from '../shared/BasicValueSpecificationEditor.js';
 
+export const IS_DRAGGABLE_FILTER_DND_TYPES_FETCHSUPPORTED = [
```
Comment (akphi) [inline packages/legend-query-builder/src/components/filter/QueryBuilderFilterPanel.tsx:88]:
```suggestion
export const IS_DRAGGABLE_FILTER_DND_TYPES_FETCH_SUPPORTED = [
```

File: packages/legend-art/src/panel/PanelDropZone.tsx (hunk 2 above)
Comment (akphi) [reply packages/legend-art/src/panel/PanelDropZone.tsx:33]: We should think about the interaction between this and the overlay, I think the border here should be on a layer that is beneath the overlay and above the content, also, it should be absolutely position to not jank the layout

File: packages/legend-art/src/panel/BlankPanelPlaceholder.tsx [hunk 7]
```diff
@@ -174,6 +176,7 @@ export const BlankPanelPlaceholder: React.FC<{
         className={clsx('blank-panel-placeholder', {
           'blank-panel-placeholder--no-click': !onClick || disabled,
           'blank-panel-placeholder--invisible': !showPlaceholder,
+          'draggable--active': draggableActive,
```
Comment (akphi) [inline packages/legend-art/src/panel/BlankPanelPlaceholder.tsx:179]: We should design this in a way that it doesn't cause the layout to jank

File: packages/legend-query-builder/style/_query-builder-filter.scss [hunk 8]
```diff
@@ -164,6 +164,11 @@
     font-weight: 700;
     color: var(--color-dark-grey-300);
     height: 2.8rem;
+
+    &--potential__dropzone {
```
Comment (akphi) [inline packages/legend-query-builder/style/_query-builder-filter.scss:168]: --> `--dropable`

Comment (akphi) [reply packages/legend-query-builder/style/_query-builder-filter.scss:168]: again, for these, we should think of a way to make them absolutely position to not jank the layout

File: packages/legend-query-builder/style/_query-builder-post-filter.scss [hunk 9]
```diff
@@ -164,6 +164,11 @@
     font-weight: 700;
     color: var(--color-dark-grey-300);
     height: 2.8rem;
+
+    &--potential__dropzone {
```
Comment (akphi) [inline packages/legend-query-builder/style/_query-builder-post-filter.scss:168]: --> `--dropable`

again, for these, we should think of a way to make them absolutely position to not jank the layout

File: packages/legend-art/style/base/_common.scss [hunk 10]
```diff
@@ -108,6 +108,30 @@
   }
 }
 
+.dnd__entry-potential__dropzone__indicator {
```
Comment (akphi) [inline packages/legend-art/style/base/_common.scss:111]: some of these styles are fine to be here, some of them better off be in `query buidler`, for example, specializations like `--full`, `--tall`, etc.

File: packages/legend-query-builder/src/components/filter/QueryBuilderFilterPanel.tsx [hunk 11]
```diff
@@ -212,81 +247,180 @@ const QueryBuilderFilterConditionEditor = observer(
       reloadValues: debouncedTypeaheadSearch,
       cleanUpReloadValues,
     };
+
+    const { isPotentiallyDragging } = useDragLayer((monitor) => ({
```
Comment (akphi) [inline packages/legend-query-builder/src/components/filter/QueryBuilderFilterPanel.tsx:251]: --> `showDropableSuggestion`

File: packages/legend-query-builder/src/components/filter/QueryBuilderFilterPanel.tsx [hunk 12]
```diff
@@ -212,81 +247,180 @@ const QueryBuilderFilterConditionEditor = observer(
...
+                      />
+                    </PanelEntryDropZonePlaceholder>
+                  </div>
+                )}
+              </div>
+            </div>
+          </>
+        ) : (
+          <>
+            {' '}
```
Comment (akphi) [inline packages/legend-query-builder/src/components/filter/QueryBuilderFilterPanel.tsx:346]: let's clean up these `{' '}` next time please

File: packages/legend-query-builder/src/components/filter/QueryBuilderFilterPanel.tsx (hunk 11 above)
Comment (akphi) [reply packages/legend-query-builder/src/components/filter/QueryBuilderFilterPanel.tsx:251]: let's find a way so we don't have to duplicate the code like this, shall we?

File: packages/legend-query-builder/src/components/fetch-structure/QueryBuilderPostFilterPanel.tsx [hunk 13]
```diff
@@ -359,80 +386,177 @@ const QueryBuilderPostFilterConditionEditor = observer(
...
-                transformOrigin: { vertical: 'top', horizontal: 'left' },
-                elevation: 7,
-              }}
-            >
-              <div className="query-builder-post-filter-tree__condition-node__operator__label">
-                {node.condition.operator.getLabel()}
-              </div>
-              <div className="query-builder-post-filter-tree__condition-node__operator__dropdown__trigger">
-                <CaretDownIcon />
+        {isPotentiallyDragging ? (
```
Comment (akphi) [inline packages/legend-query-builder/src/components/fetch-structure/QueryBuilderPostFilterPanel.tsx:406]: let's find a way so we don't have to duplicate the code like this, shall we?

**Review by akphi — CHANGES_REQUESTED** [review]

**Review by akphi — CHANGES_REQUESTED** [review]
Looks good to me, though you have some CI failures there. Also, it's a TODO that we think of the problem with DnD I told you a while back, can you file a Github issue for that? Thanks!

**Review by akphi — APPROVED** [review]

**Pull request merged**